pkg/pillar/cmd/domainmgr/containerd.go

```diff
@@ -90,70 +94,159 @@ func ctrRun(domainName, xenCfgFilename, imageHash string, envList map[string]str
 // but instead use containerd to set it up, and xl to run it, so this does not need to do anything.
 // If/when we invert it, and have containerd launch the container with a microvm wrapper,
 // *then* this will need to work.
-func ctrStop(id string, force bool) error {
-	log.Infof("ctrStop %s %t\n", id, force)
+func ctrStop(containerID string, force bool) error {
+	log.Infof("ctrStop %s %t\n", containerID, force)
 	log.Infof("ctr stop done\n")
 	return nil
 }
 
 // ctrRm remove an existing container
-func ctrRm(id string) error {
-	log.Infof("ctrRm %s\n", id)
+func ctrRm(containerID string) error {
+	log.Infof("ctrRm %s\n", containerID)
+	ctrdClient, err := getContainerdClient(ctrdSocket)
+	if err != nil {
+		log.Errorf("Could not build containerd client. %v", err.Error())
+		return fmt.Errorf("Could not build containerd client. %v", err.Error())
+	}
+	defer ctrdClient.Close()
+	ctrdCtx := getContainerdContext(CtrdServicesNamespace)
+
+	container, err := loadContainer(containerID)
```

Comment:
As below, we need to distinguish between an error vs the container does not exist. If it is gone, then we are fine, we got where we wanted, so `return nil`.

Reply:
Done.